### PR TITLE
Fix for programmatic configuration of logging in client

### PR DESCRIPTION
**Pull request by oguzdemir**
fixes https://github.com/hazelcast/hazelcast/issues/7764

### CODE REVIEW

**Comment by mustafaiman** [comment]
verify

**Comment by mustafaiman** [comment]
run-lab-run

**Comment by mustafaiman** [comment]
run-lab-run

**Comment by ihsandemir** [comment]
@oguzdemir Is there a test somewhere for verifying that system property option is still working?

**Comment by mustafaiman** [comment]
👍 after Ihsan's comment is verified.

**Comment by ihsandemir** [comment]
verify

**Comment by mustafaiman** [comment]
run-lab-run

**Comment by ihsandemir** [comment]
verify

**Comment by ihsandemir** [comment]
👍 After tests pass.

**Comment by mustafaiman** [comment]
run-lab-run

File: hazelcast-client/src/test/java/com/hazelcast/client/impl/ClientLoggerConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        props.setProperty("log4j.appender.FILE.File","./" + LOG_FILENAME);
+        props.setProperty("log4j.appender.FILE.ImmediateFlush" , "true");
+        PropertyConfigurator.configure(props);
+
+        ClientConfig config = new ClientConfig() ;
+        if (programmaticConfiguration) {
+            config.setProperty( "hazelcast.logging.type", "log4j" );
+        }
+        else {
+            System.setProperty( "hazelcast.logging.type", "log4j" );
```
Comment (eminn) [inline hazelcast-client/src/test/java/com/hazelcast/client/impl/ClientLoggerConfigurationTest.java]: Can we clear this property after the test finished ?

Comment (mustafaiman) [inline hazelcast-client/src/test/java/com/hazelcast/client/impl/ClientLoggerConfigurationTest.java]: It's there in @After method https://github.com/hazelcast/hazelcast/pull/8518/files/3dbfd49737cd9b9c5179ebf0edd9d7dcdc38a769#diff-b6637b9e63e43d28f960bce3b78b5349R35

**Pull request merged**